packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -152,29 +161,6 @@ export async function transformWithEsbuild(
     tsconfigRaw,
   } as ESBuildOptions
 
-  // esbuild uses tsconfig fields when both the normal options and tsconfig was set
-  // but we want to prioritize the normal options
-  if (
-    options &&
-    typeof resolvedOptions.tsconfigRaw === 'object' &&
-    resolvedOptions.tsconfigRaw.compilerOptions
-  ) {
-    options.jsx && (resolvedOptions.tsconfigRaw.compilerOptions.jsx = undefined)
-    options.jsxFactory &&
-      (resolvedOptions.tsconfigRaw.compilerOptions.jsxFactory = undefined)
-    options.jsxFragment &&
-      (resolvedOptions.tsconfigRaw.compilerOptions.jsxFragmentFactory =
-        undefined)
-    options.jsxImportSource &&
-      (resolvedOptions.tsconfigRaw.compilerOptions.jsxImportSource = undefined)
-    options.target &&
-      (resolvedOptions.tsconfigRaw.compilerOptions.target = undefined)
-  }
-
-  delete resolvedOptions.include
-  delete resolvedOptions.exclude
-  delete resolvedOptions.jsxInject
```

Comment:
These were needed because there is a cast here: https://github.com/vitejs/vite/pull/12493/files#diff-6d149ac9706cd508b52db0c059a0f01d670620a9a5a73cd7b122ce62b6b69471R205, from our own extended options. These fields are no longer present when calling `transformWithEsbuild` after the changes in this PR.